### PR TITLE
RET-3767: Added autoListFlag caseAdditionalSecurityFlag caseRestrictedFlag and caseInterpreterRequiredFlag

### PR DESCRIPTION
**Pull request by JackReevies**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RET-3767

### Change description ###

Added autoListFlag caseAdditionalSecurityFlag caseRestrictedFlag and caseInterpreterRequiredFlag

**Does this PR introduce a breaking change?** (check one with "x")

```
[] Yes
[x] No
```

### CODE REVIEW

**Review by Itay-Kirsh — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseFlagsController.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        if (!verifyTokenService.verifyTokenSignature(userToken)) {
+            log.error(INVALID_TOKEN, userToken);
+            return ResponseEntity.status(FORBIDDEN.value()).build();
+        }
+
+        CaseData caseData = ccdRequest.getCaseDetails().getCaseData();
+        caseFlagsService.processNewlySetCaseFlags(caseData);
+
+        // Code here
```
[the remark this answers is not in the document]
Comment (JackReevies) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseFlagsController.java]: haha my code snippet, removed now

**Comment by Itay-Kirsh** [comment]
LGTM

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java [hunk 2]
```diff
@@ -278,6 +278,7 @@ public ResponseEntity<CCDCallbackResponse> postDefaultValues(
             //create NOC answers section
             caseData = nocRespondentRepresentativeService.prepopulateOrgPolicyAndNoc(caseData);
             caseFlagsService.setupCaseFlags(caseData);
+            caseFlagsService.setDefaultFlags(caseData);
```
Comment (robin-ieong) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java:281]: unit test please

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseFlagsController.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Accessed successfully",
+            content = {
+                @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = CCDCallbackResponse.class))
+            }),
+        @ApiResponse(responseCode = "400", description = "Bad Request"),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<CCDCallbackResponse> aboutToSubmit(
```
Comment (robin-ieong) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseFlagsController.java:57]: unit test please

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseFlagsServiceTest.java [hunk 4]
```diff
@@ -43,4 +54,162 @@ void setupCaseFlags_setsShellCaseFlags() {
...
+        );
+
+        caseFlagsService.processNewlySetCaseFlags(caseData);
+
+        assertThat(caseData.getCaseInterpreterRequiredFlag(), is(YES));
+    }
+
+    @Test
+    void processNewlySetCaseFlags_respondentNeedsLanguageInterpreter_setsInterpreterRequiredFlagTrue() {
+        caseData.setClaimantFlags(CaseFlagsType.builder().details(ListTypeItem.from()).build());
```
Comment (robin-ieong) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseFlagsServiceTest.java:84]: looks like these two lines can be moved to @BeforeEach ?

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseFlagsServiceTest.java [hunk 5]
```diff
@@ -43,4 +54,162 @@ void setupCaseFlags_setsShellCaseFlags() {
...
+                GenericTypeItem.from(FlagDetailType.builder().name(LANGUAGE_INTERPRETER).status(ACTIVE).build())
+        );
+
+        caseFlagsService.processNewlySetCaseFlags(caseData);
+
+        assertThat(caseData.getCaseInterpreterRequiredFlag(), is(YES));
+    }
+
+    @Test
+    void processNewlySetCaseFlags_claimantNeedsSignLanguageInterpreter_setsInterpreterRequiredFlagTrue() {
```
Comment (robin-ieong) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseFlagsServiceTest.java:97]: Can we just have two ParameterizedTest? one for RespondentFlags and one for ClaimantFlags
Seems to be a lot of redundant test code here.

Comment (JackReevies) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseFlagsServiceTest.java:97]: No, although its a lot of repeated code, its very clear from the title what exactly the test is testing, if we start abstracting away this information, it becomes less readable. https://mtlynch.io/good-developers-bad-tests/

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java (hunk 2 above)
Comment (JackReevies) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java:281]: this already exists

Comment (robin-ieong) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java:281]: setDefaultFlags still not being verified in your test

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseFlagsServiceTest.java (hunk 5 above)
Comment (robin-ieong) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseFlagsServiceTest.java:94]: Disagree on it as discussed

**Review by robin-ieong — APPROVED** [review]

**Pull request merged**